File: src/main/java/club/mindtech/mindbot/commands/CommandPing.java
package club.mindtech.mindbot.commands;

import club.mindtech.mindbot.MindBot;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;

public class CommandPing extends BaseCommand {
    public CommandPing() {
        super("ping", "Returns the gateway ping of the user!", "ping");
    }

    @Override
    public void onSlashCommand(SlashCommandEvent event) {
        MindBot.getAPI()
               .getRestPing()
               .queue(ping -> event.deferReply().setContent("Pong! " + ping + " ms").queue());
    }
}

File: src/main/kotlin/club/mindtech/mindbot/commands/CommandPing.kt
package club.mindtech.mindbot.commands

import club.mindtech.mindbot.bot
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

class CommandPing : BaseCommand("ping", "Returns the gateway ping of the user!", "ping") {
    override fun onSlashCommand(event: SlashCommandInteractionEvent) {
        bot.jda.restPing.queue { event.deferReply().setContent("Pong! $it ms").queue() }
    }
}
